### PR TITLE
MRG, ENH: Add sphere method to decimate_surface

### LINKED ISSUE #7259: decimate_surface breaks the meshes in some cases
#### Describe the bug
Meshes reduced in size using decimate_surface are sometime invalid and crash the BEM computation generating a `RuntimeError` exception with message like `Surface inner skull has topological defects: 3 / 5119 vertices have fewer than three neighboring triangles [1381, 1382, 1383]`.

#### Steps to reproduce
The following code:
```
from mne.surface import decimate_surface, read_surface
path = "/usr/local/freesurfer/subjects/ANTS7-5Months3T/bem/inner_skull_large.surf"
from mne.surface import _triangle_neighbors

# Code extracted and slighly modified from mne.surface.complete_surface_info
# for compactness of the example
def check_topological_defect(faces, vertices):
    #    Find neighboring triangles, accumulate vertex normals, normalize
    neighbor_tri = _triangle_neighbors(faces, vertices)

    #   Check for topological defects
    zero, fewer = list(), list()
    for ni, n in enumerate(neighbor_tri):
        if len(n) < 3:
            if len(n) == 0:
                zero.append(ni)
            else:
                fewer.append(ni)
    if len(zero) > 0:
        print('    Vertices do not have any neighboring '
                    'triangles: [%s]' % ', '.join(str(z) for z in zero))
    if len(fewer) > 0:
        print('    Vertices have fewer than three neighboring '
                    'triangles, removing neighbors: [%s]'
                    % ', '.join(str(f) for f in fewer))

vertices, faces = read_surface(path)

print("Checking the full mesh:")
check_topological_defect(faces, len(vertices))

vertices, faces = decimate_surface(vertices, faces, n_triangles=5120)

print("Checking the decimated mesh:")
check_topological_defect(faces, len(vertices))
```
Produces the following output:
```
Checking the full mesh:
Checking the decimated mesh:
    Vertices have fewer than three neighboring triangles, removing neighbors: [1381, 1382, 1383]
```
I can send inner_skull_large.surf by email for you to reproduce this issue.

#### Expected results
No vertices with fewer than three neighbouring triangles should be introduced by decimate_surface.

#### Additional information
```
Platform:      Linux-5.3.0-26-generic-x86_64-with-debian-buster-sid
Python:        3.7.4 (default, Aug 13 2019, 20:35:49)  [GCC 7.3.0]
Executable:    /home/christian/anaconda3/bin/python
CPU:           x86_64: 12 cores
Memory:        31.3 GB

mne:           0.20.dev0
numpy:         1.17.4 {blas=mkl_rt, lapack=mkl_rt}
scipy:         1.4.1
matplotlib:    3.1.2 {backend=Qt5Agg}

sklearn:       0.22.1
numba:         Not found
nibabel:       3.0.0
cupy:          Not found
pandas:        0.25.3
dipy:          Not found
mayavi:        4.6.2 {qt_api=pyqt5, PyQt5=5.14.1}
pyvista:       Not found
vtk:           8.1.2
```

### PR DESCRIPTION
**Pull request by larsoner**
Closes #7259

@christian-oreilly can you see how well `method='sphere'` works once you have a suitable (not broken, no holes, etc.) high-res mesh?

### CODE REVIEW

**Comment by larsoner** [comment]
@Davi1990 you can also see if this helps you make a mesh from #7232

**Comment by christian-oreilly** [comment]
> @christian-oreilly can you see how well method='sphere' works once you have a suitable (not broken, no holes, etc.) high-res mesh?

@larsoner Yep, I'll check it shortly.

**Comment by agramfort** [comment]
@christian-oreilly any chance you can check and report back here if it's good to go from your end?

**Comment by larsoner** [comment]
BTW @agramfort this is part of what we'll need for #6784 so feel free to look / try / merge if you're happy

File: mne/surface.py [hunk 1]
```diff
@@ -1048,16 +1093,50 @@ def decimate_surface(points, triangles, n_triangles):
...
     points : ndarray
         The decimated points.
     triangles : ndarray
         The decimated triangles.
+
+    Notes
+    -----
+    **"quadric" mode**
+
+    This requires VTK. If an if an odd target number was requested,
```
Comment (agramfort) [inline mne/surface.py]: If an?

File: mne/surface.py [hunk 2]
```diff
@@ -1048,16 +1093,50 @@ def decimate_surface(points, triangles, n_triangles):
...
+
+    **"sphere" mode**
+
+    This requires Freesurfer to be installed and available in the
+    environment. The destination number of triangles must be one of
+    ``[20, 80, 320, 1280, 5120, 20480]`` for ico (0-5) downsampling or one of
+    ``[8, 32, 128, 512, 2048, 8192, 32768]`` for oct (1-7) downsampling.
+
+    This mode is slower, but could be more suitable for decimating meshes for
+    BEM creation (recommended ``n_tringles=5120``) due to better topological
```
Comment (agramfort) [inline mne/surface.py]:
```suggestion
    BEM creation (recommended ``n_triangles=5120``) due to better topological
```

**Pull request merged**